src/Course/List.hs

```diff
@@ -91,8 +91,7 @@ headOr =
 product ::
   List Int
   -> Int
-product =
-  error "todo: Course.List#product"
+product xs = foldRight (*) 1 xs
```

Comment:
alternate:

```
product = foldLeft (*) 1
```

(ie point-free form, but also, foldLeft is if you want it to run in constant space) `foldRight` is better if you want laziness tho. 👍